src/mpi/init/initthread.c

```diff
@@ -431,6 +431,7 @@ int MPIR_Init_thread(int *argc, char ***argv, int required, int *provided)
     MPIR_Process.attrs.io = MPI_PROC_NULL;
     MPIR_Process.attrs.lastusedcode = MPI_ERR_LASTCODE;
     MPIR_Process.attrs.tag_ub = MPIR_TAG_USABLE_BITS;
+    MPIR_Process.attrs.tag_bits = MPID_TAG_BITS;
```

Comment:
I'm a little confused by this. So we set the tag_bits value to the device defined value `MPID_TAG_BITS`, but later on, the device defines `MPID_TAG_BITS` to be this global variable? There's got to be a clearer way of defining this...

Reply:
If I am not mistaken this should be the default (i.e., 31). The netmod should overwrite these values later on. I think this is a mistake `MPIR_Process.attrs.tab_bits` is set by the device to default (see `src/mpid/ch4/src/ch4_init.c`). So both `MPIR_Process.attrs.tag_ub` and `MPIR_Process.attrs.tag_bits` should be assigned in `MPID_Init` to default and then overwritten, if necessary, by the netmod. This must be an error I did when reintroducing the `tag_ub` in MPIR_Process after removing it. Sorry for the confusion

Reply:
To be clear, I think it's perfectly fine to store `tag_ub` in the `MPIR_Process.attrs` object. We should set it once (based on the number of tag bits supported by the device - the number of reserved bits) and reuse it when needed.